plugins/iommu/fu-plugin-iommu.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-hash.h"
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-iommu-device.h"
+
+struct FuPluginData {
+	gboolean		 has_iommu;
+};
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_add_udev_subsystem (plugin, "iommu");
+	fu_plugin_set_device_gtype (plugin, FU_TYPE_IOMMU_DEVICE);
+}
+
+void
+fu_plugin_device_added (FuPlugin *plugin, FuDevice *dev)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	data->has_iommu = TRUE;
+}
+
+void
+fu_plugin_add_security_attrs (FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	/* create attr */
+	attr = fwupd_security_attr_new ("org.kernel.IOMMU");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_set_name (attr, "Input/Output Memory Management Unit");
```

Comment:
I think `IOMMU` is a much better name fwiw. We don't expand TPM or SPI so it just looks weird.

Reply:
fixed